src/component/Weather/Weather.jsx

```diff
@@ -0,0 +1,88 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+import "./weather.css";
+
+const Weather = () => {
+  const [coordinates, setCoordinates] = useState({ latitude: 0, longitude: 0 });
+  const [weather, setWeather] = useState(null);
+  const [city, setCity] = useState(localStorage.getItem("city"));
+  const [isInputVisible, setIsInputVisible] = useState(false);
+
+  const getWeatherData = async () => {
+    try {
+      const response =
+        coordinates.latitude !== 0 &&
+        (await axios.get(
+          `https://api.openweathermap.org/data/2.5/weather?${
```

Comment:
Suggestion: You could keep the APIs in a separate file

Reply:
ok will try to do this next pr